app/controllers/devise_token_auth/sessions_controller.rb

```diff
@@ -5,19 +5,37 @@ class SessionsController < DeviseTokenAuth::ApplicationController
 
     def create
       # honor devise configuration for case_insensitive_keys
-      if resource_class.case_insensitive_keys.include?(:email)
-        email = resource_params[:email].downcase
-      else
-        email = resource_params[:email]
-      end
+      if resource_params.include?(:email)
+        q_value = resource_params[:email]
+
+        if resource_class.case_insensitive_keys.include?(:email)
+          q_value = resource_params[:email].downcase
+        end
+
+        q = "uid = ? AND provider='email'"
 
-      q = "uid = ? AND provider='email'"
+        if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
+          q = "BINARY uid = ? AND provider='email'"
+        end
 
-      if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
-        q = "BINARY uid = ? AND provider='email'"
+      elsif resource_params.include?(:username)
```

Comment:
By default, you just have 1 key (:email). What happened if you have 3 keys ? 

Should I check if there's one authentication_keys on the params ?